### PR TITLE
Support for NSData, NSMutableData test attachments with graceful ignore for unknown types

### PR DESCRIPTION
**Pull request by diegoperini**
Builds one top of https://github.com/danielpaulus/go-ios/pull/363 since attachments are almost always received via fragmented DTX messages.

### CODE REVIEW

File: ios/dtx_codec/encoder.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func writeHeader(messageBytes []byte, messageLength uint32, Identifier int, Conv
 	binary.LittleEndian.PutUint32(messageBytes[28:], expectsReplyUint32)
 }
 
-func writePayloadHeader(messageBytes []byte, MessageType int, payloadLength int, auxLength int) {
+func writePayloadHeader(messageBytes []byte, MessageType MessageType, payloadLength int, auxLength int) {
 	binary.LittleEndian.PutUint32(messageBytes, uint32(MessageType))
```
Comment (danielpaulus) [inline ios/dtx_codec/encoder.go:78]: @diegoperini  why don't you make it an alias for uint32 and stop the casting?

**Review by danielpaulus — APPROVED** [review]
Maybe consider changing the MessageType to uint32. looks good otherwise

File: ios/dtx_codec/encoder.go (hunk 1 above)
Comment (diegoperini) [reply ios/dtx_codec/encoder.go:78]: We define a function on top of `MessageType` which maps numbers to human friendly names under the `String()` function. I'm kinda new to golang but I believe we cannot do that on type aliases. See here: https://github.com/danielpaulus/go-ios/pull/364/files#diff-e74aeb8496b7de791c62636d22da05ac8451409e098d70365e3c3d84a1cf48afR76

**Comment by diegoperini** [comment]
> Maybe consider changing the MessageType to uint32. looks good otherwise

Done

**Pull request merged**